### PR TITLE
Implementing bed changes in issue #56

### PR DESCRIPTION
**Pull request by TomLivesey-UoM**
Implemented the bullet points found in issue #56 - improvements to bed files after discussion w/ users.

### CODE REVIEW

**Review by egle0711 — APPROVED** [review]
Well-implemented, with clear and understandable changes and logical justifications for each. 

Changes add flexibility to the BED file generation process and improvements increase the robustness of the code. 

Consistent with good coding practices, don't see any issues that would prevent this pull request from being merged.

**Comment by jessicakan789** [comment]
When I ran python main.py I got this error:

  File "main.py", line 73
    filename = f"{formatted_panel_name}_{genome_build}_{date_str}.bed"
                                                                     ^
SyntaxError: invalid syntax

**Review by jessicakan789 — APPROVED** [review]

File: PanelSearch/main.py
Comment (jessicakan789) [inline PanelSearch/main.py:1]: For the BED file generation do we want to allow the user to input "y" and "Y"?

**Comment by jessicakan789** [comment]
If you would like to search by R-code, enter 1. If you would like to enter a disease description, enter 2.
1
Enter your search term: (e.g., R128 or pneumothorax)
R128
Which genome build would you like to use? Enter 1 for GRCh37. Enter 2 for GRCh38.
1
2023-12-21 08:49:48,003 INFO: Starting PanelSearch with input type: R-code
GRch37 build selected.
Panel ID: 13
Panel Name: Brugada syndrome and cardiac sodium channel disease
GMS Sign-off Status: Signed Off
Associated NGTD R-codes: R128
Gene Number: 23
Gene List:
HGNC_ID  Symbol  Coords  Mane_Select_Transcript
HGNC:10593 SCN5A  3:38589548-38691164 NM_000335.5
HGNC:6251 KCNH2  7:150642049-150675403 NM_000238.4
HGNC:60 ABCC9  12:21950335-22094336 NM_020297.4
HGNC:493 ANK2  4:113739265-114304896 NM_001148.6
HGNC:1390 CACNA1C  12:2079952-2802108 NM_000719.7
HGNC:1399 CACNA2D1  7:81575760-82073114 NM_000722.4
HGNC:1402 CACNB2  10:18429606-18830798 NM_201596.3
HGNC:1529 CAV3  3:8775486-8883492 NM_033337.3
HGNC:2900 DLG1  3:196769431-197026171 NM_001366207.1
HGNC:28956 GPD1L  3:32147181-32210205 NM_015141.4
HGNC:16882 HCN4  15:73612200-73661605 NM_005477.3
HGNC:6239 KCND3  1:112313284-112531777 NM_001378969.1
HGNC:6243 KCNE3  11:74165886-74178774 NM_005472.5
HGNC:6241 KCNE5  X:108866929-108868393 NM_012282.4
HGNC:6269 KCNJ8  12:21917889-21928515 NM_004982.4
HGNC:9024 PKP2  12:32943679-33049774 NM_001005242.3
HGNC:17679 RANGRF  17:8191815-8193410 NM_016492.5
HGNC:10582 SCN10A  3:38738293-38835501 NM_006514.4
HGNC:10586 SCN1B  19:35521588-35531352 NM_001037.5
HGNC:10589 SCN2B  11:118032666-118047388 NM_004588.5
HGNC:20665 SCN3B  11:123499895-123525952 NM_001040151.2
HGNC:16643 SLMAP  3:57741177-57914895 NM_001377540.1
HGNC:17993 TRPM4  19:49660998-49715093 NM_017636.4
2023-12-21 08:49:50,698 INFO: Panel data processed successfully
Generate BED file? (Y/N) 
Y
For exon coordinates on transcript, press '1'. For genomic coordinates, press '2'. 
1
Standard +/- 5 bp padding used in beds file for exon - enter a number between 0-15 to change this. Otherwise press enter.10
2023-12-21 08:50:14,355 INFO Using a padding value of +/- 10 bp.
2023-12-21 08:50:14,358 DEBUG JSON data parsed successfully.
2023-12-21 08:50:14,363 INFO BED and JSON files written successfully.
2023-12-21 08:50:14,368 INFO: BED file generation initiated
Would you like to save this search against a patient ID? (Y/N) 
Y
What patient ID would you like to save this search against? 
testeroo
Brugada syndrome and cardiac sodium channel disease
GRch37
Traceback (most recent call last):
  File "/home/jessicak/PycharmProjects/2023-Software-Development-Assessment/PanelSearch/main.py", line 148, in <module>
    main()
  File "/home/jessicak/PycharmProjects/2023-Software-Development-Assessment/PanelSearch/main.py", line 135, in main
    create_sql_record(panel_name,SEARCH.genome_build, pid)
  File "/home/jessicak/PycharmProjects/2023-Software-Development-Assessment/PanelSearch/main.py", line 81, in create_sql_record
    PK_Parse_Data_to_SQL_cloud(pid,genome_build,PK = panel_name)
  File "/home/jessicak/PycharmProjects/2023-Software-Development-Assessment/PanelSearch/API_to_SQL_cloud.py", line 11, in PK_Parse_Data_to_SQL_cloud
    result = PanelApp_Request_Parse.panelapp_search_parse(response.json(), genome_build)
NameError: name 'PanelApp_Request_Parse' is not defined. Did you mean: 'panelapp_search_parse'?



^ error fixed in logging enhancement branch

**Pull request merged**